cpp/src/arrow/flight/flight_test.cc

```diff
@@ -1247,7 +1247,9 @@ TEST_F(TestBasicAuthHandler, CheckPeerIdentity) {
   ASSERT_EQ(result->body->ToString(), "user");
 }
 
-TEST_F(TestTls, DoAction) {
+// ARROW-7701: this test is flaky on MacOS and segfaults (due to gRPC
+// bug?)
+TEST_F(TestTls, DISABLED_DoAction) {
```

Comment:
You could instead do this, to avoid skipping the test on other platforms:
```c++
#ifdef __APPLE__
  // ARROW-7701: this test is flaky on MacOS and segfaults (due to gRPC
  // bug?)
  return;
#endif
```

Reply:
Fair enough, I should've looked around harder. I changed this to an ifdef guard.